code/iaas/service-discovery/server/src/main/java/io/cattle/platform/servicediscovery/deployment/impl/DeploymentUnit.java

```diff
@@ -194,6 +194,10 @@ public boolean isComplete() {
     protected Map<String, String> getLabels(DeploymentUnitInstance instance) {
         Service service = instance.getService();
         Map<String, String> labels = new HashMap<>();
+
+        // TODO: This might change in the future if we merge the labels from all services
+        labels.putAll(context.sdService.getServiceLabels(service));
```

Comment:
But I see why you need these labels here as you operate on them later in the code. So should be fine to leave it here for now.